test/OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.test.cpp

```diff
@@ -0,0 +1,422 @@
+using ostk::astrodynamics::trajectory::LocalOrbitalFrameFactory;
+using ostk::astrodynamics::trajectory::State;
+using ostk::astrodynamics::trajectory::state::CoordinateSubset;
+using ostk::astrodynamics::trajectory::state::coordinatesubset::CartesianPosition;
+using ostk::astrodynamics::trajectory::state::coordinatesubset::CartesianVelocity;
+using ostk::astrodynamics::trajectory::StateBuilder;
+
+class OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver : public ::testing::Test
+{
+   protected:
+    void SetUp() override
+    {
+        // Define initial state (position and velocity)
+        const Position initialPosition = Position::Meters({1.0, 0.0, 0.0}, Frame::GCRF());
+        const Velocity initialVelocity = Velocity::MetersPerSecond({0.0, 0.0, 0.0}, Frame::GCRF());
+        trueState_ = {Instant::J2000(), initialPosition, initialVelocity};
+
+        // Define observations (true states with noise)
+        const Size observationCount = 100;
+        const Real timeStep = 0.1;  // seconds
+
+        for (Size i = 0; i < observationCount; ++i)
+        {
+            const Real time = timeStep * (Real)i;
+            const Real position = std::cos(time);
+            const Real velocity = -std::sin(time);
+
+            const Position truePosition = Position::Meters({position, 0.0, 0.0}, Frame::GCRF());
+            const Velocity trueVelocity = Velocity::MetersPerSecond({velocity, 0.0, 0.0}, Frame::GCRF());
+
+            // Add noise to the true state
+            std::mt19937 rng(42);  // Fixed seed for reproducibility
+            std::uniform_real_distribution<double> dist(-0.01, 0.01);
+            const Vector3d noisePositionCoordinates = {dist(rng), 0.0, 0.0};
+            const Vector3d noiseVelocityCoordinates = {dist(rng), 0.0, 0.0};
+
+            const Position combinedPosition =
+                Position::Meters(truePosition.getCoordinates() + noisePositionCoordinates, Frame::GCRF());
+            const Velocity combinedVelocity =
+                Velocity::MetersPerSecond(trueVelocity.getCoordinates() + noiseVelocityCoordinates, Frame::GCRF());
+
+            observations_.add(State(Instant::J2000() + Duration::Seconds(time), combinedPosition, combinedVelocity));
+        }
+
+        const Position perturbedPosition = Position::Meters({0.1, 0.0, 0.0}, Frame::GCRF());
+        const Velocity perturbedVelocity = Velocity::MetersPerSecond({0.1, 0.0, 0.0}, Frame::GCRF());
+        initialGuessState_ = trueState_ + State(trueState_.getInstant(), perturbedPosition, perturbedVelocity);
+
+        positionSigmas_ << 0.1, 0.1, 0.1;
+        velocitySigmas_ << 0.1, 0.1, 0.1;
+
+        observationSigmas_ = {
+            {*CartesianPosition::Default(), positionSigmas_}, {*CartesianVelocity::Default(), velocitySigmas_}
+        };
+
+        initialStateSigmas_ = {
+            {*CartesianPosition::Default(), positionSigmas_}, {*CartesianVelocity::Default(), velocitySigmas_}
+        };
```

Comment:
Needs to match the sigma specified in the normal distribution, and the initial guess sigmas should be different than the observation sigmas, so that you can actually test the weighting properly